src/engine/src/scheduler/IScheduler.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <entt/entt.hpp>
+
+namespace ES::Engine {
+class Registry;
+}
+
+namespace ES::Engine::Scheduler {
+using USystem = std::function<void(Registry &)>;
+
+/**
+ * @brief Interface to be implemented for every schedulers
+ */
+class IScheduler {
+  public:
+    explicit IScheduler(Registry &registry) : _registry(registry) {}
+    virtual void RunSystems(std::vector<USystem> systems) = 0;
```

Comment:
Can you add documentation to `RunSystems` pls

Reply:
done